CHANGELOG.md

```diff
@@ -6,6 +6,7 @@ Mapbox welcomes participation and contributions from everyone.
 #### Features
 
 #### Bug fixes and improvements
+- Introduced `RoutesSetCallback` parameter to `MapboxNavigation#setNavigationRoutes`, which is called after the routes are set or they have been declared invalid. If invalid, they will not be passed to `RoutesObserver#onRoutesChanged` [#5945](https://github.com/mapbox/mapbox-navigation-android/pull/5945).
```

Comment:
> If invalid, they will not be passed to `RoutesObserver#onRoutesChanged`

I think we should pass empty routes. Consider following scenario.
```kotin
val validRoute = createValidRoute()
val invalidRoute = createInvalidRoute()

val latestNavigationRoutes: List<NavigationRoute> = emptyList()
mapboxNavigation.registerRouteProgressObserver { result ->  latestNavigationRoutes = result.navigationRoutes }

mapboxNavigation.setNavigationRoutes(validRoutes)
// wait for processing
mapboxNavigation.setNavigationRoutes(invalidRoutes)
// wait for processing

assertEquals(emptyList(), latestNavigationRoutes)
assertEquals(emptyList(), mapboxNavigation.getNavigationRoutes())
```

Reply:
BTW, this suggestion is based on assumption that native navigator switches to a free drive after setting routes with error.

Reply:
NN tracks old route in case of error. The suggestion isn't relevant.

Reply:
Then it makes sense to not pass anything.